src/data_structures/interval_tree.rs

```diff
@@ -272,13 +255,24 @@ fn swap_interval_data<N, D>(node_1: &mut Node<N, D>, node_2: &mut Node<N, D>) {
 }
 
 fn intersect<N: Ord>(range_1: &Range<N>, range_2: &Range<N>) -> bool {
-    range_1.end > range_2.start && range_1.start < range_2.end
+    range_1.start < range_1.end && range_2.start < range_2.end &&
```

Comment:
If the Range is not `pub`, you will only be able to use the `::new` method. There is a macro that allows to easily support dereferencing the newtype into the wrapped type. I use it [here](https://github.com/rust-bio/rust-bio/blob/master/src/stats/probs/mod.rs#L39).